File: Rudder/Classes/ScreenPropertyBuilder.m
//
//  ScreenPropertyBuilder.m
//  RudderSDKCore
//
//  Created by Arnab Pal on 17/10/19.
//  Copyright © 2019 Rudderlabs. All rights reserved.
//

#import "ScreenPropertyBuilder.h"
#import "RudderLogger.h"

@implementation ScreenPropertyBuilder

- (instancetype)setScreenName:(NSString *)screenName {
    if (self->property == nil) {
        self->property = [[RudderProperty alloc] init];
    }
    [self->property put:@"name" value:screenName];
    return self;
}

- (RudderProperty *)build {
    if (self->property == nil) {
        [RudderLogger logError:@"screen name is not set. returning blank"];
        self->property = [[RudderProperty alloc] init];
    }
    return self->property;
}

@end

File: RudderSampleProject/RudderSdkCore/ScreenPropertyBuilder.swift
//
//  ScreenPropertyBuilder.swift
//  RudderSdkCore
//
//  Created by Arnab Pal on 11/10/19.
//  Copyright © 2019 Arnab Pal. All rights reserved.
//

import Foundation

public class ScreenPropertyBuilder : NSObject {
    private var name: String? = nil
    
    public func setName(name: String) -> ScreenPropertyBuilder {
        self.name = name
        return self
    }
    
    public func build() -> Dictionary<String, NSObject>? {
        if (self.name == nil || self.name!.isEmpty) {
            RudderLogger.logError(message: "name can not be nil or empty")
            return nil
        }
        var property: Dictionary<String, NSObject> = Dictionary()
        property["name"] = self.name! as NSObject
        return property
    }
}
